torchao/quantization/quant_primitives.py

```diff
@@ -8,15 +8,24 @@
 from typing import List, Optional, Tuple, Dict, Callable, Union
 import torch, math
 
-
 from torchao.kernel.intmm import int_scaled_matmul
 from torchao.kernel.intmm import safe_int_mm
 from torchao.utils import (
     TORCH_VERSION_AT_LEAST_2_3,
     TORCH_VERSION_AT_LEAST_2_5,
 )
 from torchao.utils import _register_custom_op
+from torchao.float8.float8_tensor import (
+    LinearMMConfig,
+    GemmInputRole,
+    Float8Tensor,
+)
 
+from torchao.float8.float8_scaling_utils import (
+    hp_tensor_to_float8_dynamic,
```

Comment:
this is not a public API of `torchao.float8`, would be great to not use it here.  This function is an implementation detail and can change at any time, without any bc guarantees.